pep-0590.rst

```diff
@@ -48,145 +48,212 @@ The function pointer type
 Calls are made through a function pointer taking the following parameters:
 
 * ``PyObject *callable``: The called object
-* ``Py_ssize_t n``: The number of arguments plus an optional offset flag for the first argument in vector.
+* ``Py_ssize_t n``: The number of arguments plus the optional flag ``PY_VECTORCALL_PREPEND`` (see below)
 * ``PyObject **args``: A vector of arguments
-* ``PyTupleObject *kwnames``: A tuple of the names of the named arguments.
+* ``PyObject *kwnames``: Either ``NULL`` or a non-empty tuple of the names of the keyword arguments
 
 This is implemented by the function pointer type:
-``typedef PyObject *(*vectorcall)(PyObject *callable, Py_ssize_t n, PyObject** args, PyTupleObject *kwnames);``
+``typedef PyObject *(*vectorcallfunc)(PyObject *callable, Py_ssize_t n, PyObject **args, PyObject *kwnames);``
 
-Changes to the ``PyTypeObject``
--------------------------------
+Changes to the ``PyTypeObject`` struct
+--------------------------------------
 
-The unused slot ``printfunc tp_print`` is replaced with ``tp_vectorcall_offset``. It has the type ``uintptr_t``.
+The unused slot ``printfunc tp_print`` is replaced with ``tp_vectorcall_offset``. It has the type ``Py_ssize_t``.
```

Comment:
Specifically, we can use [some C99 features](https://www.python.org/dev/peps/pep-0007/#c-dialect). But `stdint.h`'s `uintptr_t` is one of them.
Older code doesn't use it because the C99 features were allowed relatively recently.

Reply:
In which sense is `uintptr_t` more "correct" than `Py_ssize_t`? Given that all other offset values (for example `tp_weaklistoffset`) use `Py_ssize_t`, there should be a good reason to deviate from that.

Reply:
To be pedantic, neither `uintptr_t` nor `Py_ssize_t` is correct.  The C standard says you're not allowed to store the address of a function in a variable not declared as a pointer to a function.  We do it all the time, of course.  But it would do my heart some good to not add *new* places where we bend the spec.

Reply:
It is not an address, but an offset. 
However, as we want to minimize the chance of breakage and the offset will always be less than `UINTPTR_MAX`, `uintptr_t` is better than `offset_t`.

Reply:
`Py_ssize_t` exists only because our C code had to be C89 compliant.

Reply:
To be even more pedantic, wouldn't `ptrdiff_t` be better? It's specifically meant for differences between pointers (i.e. offsets).

Reply:
FYI: according to https://en.cppreference.com/w/c/types/offsetof the resulting type of `offsetof()` is a `size_t`, so using `size_t` for `tp_vectorcall_offset` can't be wrong. And CPython typically uses `Py_ssize_t` instead of `size_t` for sizes, so `Py_ssize_t` is not that wrong either.

I don't want further bikeshedding on this, so please suggest something and I'll make the change.